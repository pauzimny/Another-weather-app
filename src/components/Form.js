import React from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Search from "@material-ui/icons/Search";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: "50%",
    alignSelf: "flex-end",
    color: "#ff680c"
  },
  input: {
    "&::after": {
      borderBottom: "3px solid #ff680c"
    }
  },

  addButton: {
    backgroundColor: "#ff670e",
    marginTop: "20px",
    color: "#fff",
    "&:hover": {
      backgroundColor: "#960c27"
    }
  }
}));

const Form = props => {
  const classes = useStyles();
  return (
    <form className="form" onSubmit={props.submit}>
      <TextField
        id="standard-with-placeholder"
        label="Nazwa miasta"
        placeholder=""
        className={classes.textField}
        margin="normal"
        value={props.value}
        onChange={props.change}
        InputProps={{ className: classes.input }}
        InputLabelProps={{
          style: {
            color: "#ff680c"
          }
        }}
      />
      <Button
        variant="contained"
        size="medium"
        className={classes.addButton}
        onClick={props.submit}
      >
        Dodaj miasto
        <Search style={{ marginLeft: "5px" }} />
      </Button>
    </form>
  );
};

export default Form;
