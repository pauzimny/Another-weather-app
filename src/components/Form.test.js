import React from "react";
import Form from "./Form.js";
import { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

it("renders correctly", () => {
  const wrapper = shallow(
    <Form onSubmit={() => {}} value="london" onChange={() => {}} />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders correctly again", () => {
  const wrapper = mount(
    <Form onSubmit={() => {}} value="london" onChange={() => {}} />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
