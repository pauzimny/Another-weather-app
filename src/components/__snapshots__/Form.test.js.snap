// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="form"
>
  <input
    className="form__input"
    placeholder=" Nazwa miasta"
    type="text"
    value="london"
  />
  <button
    className="form__button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="search"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "paddingRight": "5px",
        }
      }
      symbol={false}
      title=""
      transform={null}
    />
    Dodaj miasto
  </button>
</form>
`;

exports[`renders correctly again 1`] = `
<Form
  onChange={[Function]}
  onSubmit={[Function]}
  value="london"
>
  <form
    className="form"
  >
    <input
      className="form__input"
      placeholder=" Nazwa miasta"
      type="text"
      value="london"
    />
    <button
      className="form__button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="search"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        style={
          Object {
            "paddingRight": "5px",
          }
        }
        symbol={false}
        title=""
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-search fa-w-16 "
          data-icon="search"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={
            Object {
              "paddingRight": "5px",
            }
          }
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon>
      Dodaj miasto
    </button>
  </form>
</Form>
`;
